### DISCUSSION TITLE
Multilingual Strategy

### DISCUSSION THREAD

**Discussion by Merino** [puckeditor/puck#190] [Q&A] [answered]
We are trying to implement this project in our page builder. It is perfect for people to design their page/emails.

I'm curious how developers use this tool for multi-language pages.
Often you want to keep the design the same, but you want to translate the content.

I haven't figured out yet how to present this to the user.

Do you let the user manage each translated page in a specific language or do you let each component render in a specific language? This way the component is responsible for the translation.

Curious about your answers?

**Comment by chrisvxd** [comment, marked as answer]
Apologies for the delay in response, @Merino. I missed this one.

There are probably several ways to achieve this:

1. Off-load the multilingual concerns to an external CMS and load content via the [`external` field type](https://puckeditor.com/docs/integrating-puck/external-data-sources)
2. Implement a [custom fields](https://puckeditor.com/docs/extending-puck/custom-fields) which sets an object for each translatable string. i.e. it would show multiple inputs to the user (which they could maybe toggle with tabs or buttons) and save the data like `{ en: "Hello, world", "de": "Hallo, Welt" }`
3. Something with a [plugin](https://puckeditor.com/docs/extending-puck/plugins), which could possible connect with the custom fields option in *2* to enable a language switcher at the top of the fields.

Hope that helps!
